### PR TITLE
feat(sdk-core): Add support for BitGo Link

### PR DESCRIPTION
**Pull request by mcalcano-bg**
Ticket: POL-64

### CODE REVIEW

File: modules/bitgo/test/v2/unit/link/linkWalletNonTSS.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        .post(`/api/external-exchange/v1/enterprise/${enterprise.id}/${ethLinkWallet.coin}/wallets/${ethLinkWallet.walletId}/transfers/${transferRequest.id}/send`,
+          _.matches(signed))
+        .reply(200, transferRequest);
+
+      const stakingTransaction = await ethLinkWallet.buildSignAndSend(
+        { walletPassphrase: walletPassphrase },
+        transferRequest,
+      );
+
+      stakingTransaction.should.deepEqual(transferRequest);
```
Comment (derickson-bitgo) [inline modules/bitgo/test/v2/unit/link/linkWalletNonTSS.ts:135]: `stakingTransaction`

File: modules/bitgo/test/v2/fixtures/link/linkWallet.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { TransferRequest, TransferStatus, StakingRequest, StakingTransaction } from '@bitgo/sdk-core';
+
+export default {
+  txRequestId: '55ba0198-0b1f-44f6-94fe-8d782d633dde',
+  stakingRequest: function(transactions: StakingTransaction[]) : StakingRequest {
```
Comment (derickson-bitgo) [inline modules/bitgo/test/v2/fixtures/link/linkWallet.ts:5]: Is the StakingRequest needed here?

**Review by derickson-bitgo — APPROVED** [review]

File: modules/bitgo/test/v2/fixtures/link/linkWallet.ts (hunk 2 above)
Comment (mcalcano-bg) [reply modules/bitgo/test/v2/fixtures/link/linkWallet.ts]: Fixed.

File: modules/bitgo/test/v2/unit/link/linkWalletNonTSS.ts (hunk 1 above)
Comment (mcalcano-bg) [reply modules/bitgo/test/v2/unit/link/linkWalletNonTSS.ts]: Fixed.

**Review by derickson-bitgo — DISMISSED** [review]

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  /**
+   * Send the signed staking transaction
+   * @param signedTransaction
+   */
+  async send(signedTransaction: LinkSignedTransaction): Promise<TransferRequest> {
+    const transferId = signedTransaction.transfer.id;
+    const enterpriseId = this.wallet.toJSON().enterprise;
+    const coin = this.wallet.coin();
+    const walletId = this.wallet.id();
+    return await this.bitgo
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:121]:
```suggestion
    return this.bitgo
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        keychain: keychain[0],
+      }),
+    };
+  }
+
+  /**
+   * Send the signed staking transaction
+   * @param signedTransaction
+   */
+  async send(signedTransaction: LinkSignedTransaction): Promise<TransferRequest> {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:116]:
```suggestion
 send(signedTransaction: LinkSignedTransaction): Promise<TransferRequest> {
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        txPrebuild: stakingPrebuildTransaction.result,
+        walletPassphrase: signOptions.walletPassphrase,
+        keychain: keychain[0],
+      }),
+    };
+  }
+
+  /**
+   * Send the signed staking transaction
+   * @param signedTransaction
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:114]:
```suggestion
   * @param Promise<signedTransaction>
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+          ],
+        }),
+      };
+    }
+  }
+
+  /**
+   * Sign the transfer transaction
+   * @param signOptions
+   * @param stakingPrebuildTransaction
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:91]:
```suggestion
   * @param Promise<stakingPrebuildTransaction>
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+      .send(signedTransaction.signed)
+      .result();
+  }
+
+  /**
+   * Build, sign and send the transfer transaction.
+   * @param signOptions
+   * @param transfer
+   */
+  async buildSignAndSend(signOptions: LinkSignOptions, transfer: TransferRequest): Promise<TransferRequest> {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:132]:
```suggestion
 buildSignAndSend(signOptions: LinkSignOptions, transfer: TransferRequest): Promise<TransferRequest> {
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    return await this.build(transfer)
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:133]:
```suggestion
    return this.build(transfer)
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+    return await this.bitgo
+      .post(this.bitgo.microservicesUrl(this.sendTransferURL(enterpriseId, transferId, coin, walletId)))
+      .send(signedTransaction.signed)
+      .result();
+  }
+
+  /**
+   * Build, sign and send the transfer transaction.
+   * @param signOptions
+   * @param transfer
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:130]:
```suggestion
   * @param Promise<transfer>
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+   */
+  async buildSignAndSend(signOptions: LinkSignOptions, transfer: TransferRequest): Promise<TransferRequest> {
+    return await this.build(transfer)
+      .then((result: LinkPrebuildTransactionResult) => this.sign(signOptions, result))
+      .then((result: LinkSignedTransaction) => this.send(result));
+  }
+
+  private async createTransferRequest({
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:138]:
```suggestion
  private createTransferRequest({
```

File: modules/sdk-core/src/bitgo/link/linkWallet.ts [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+    amount,
+    connectionId,
+  }: {
+    enterpriseId: string;
+    walletId: string;
+    coin: string;
+    amount: string;
+    connectionId: string;
+  }): Promise<TransferRequest> {
+    return await this.bitgo
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:151]:
```suggestion
    return this.bitgo
```

Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/link/linkWallet.ts:151]:
```suggestion
    return this.bitgo
```

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**